internal/logger/logger.go

```diff
@@ -0,0 +1,56 @@
+package logger
+
+import (
+	"log"
+)
+
+// Logger interface for standard log library.
+type Logger interface {
+	Debugf(format string, args ...interface{})
+
+	Infof(format string, args ...interface{})
+
+	Warningf(format string, args ...interface{})
+
+	Errorf(format string, args ...interface{})
+
+	Fatalf(format string, args ...interface{})
+
+	Panicf(format string, args ...interface{})
+}
+
+type standardLogger struct {
+}
+
+// returns new looger instance. Default is standard log library.
+func NewLogger(logger Logger) Logger {
```

Comment:
_Just a general bit of feedback on this function (even though I generally would like you to steer clear of this entirely based on my other comments... the comment is for general code style feedback):_

It's really unclear as to what the contract with the caller is for this code since it seems to be doing too many things.  The function takes a Logger which may or may not be nil.  If it isn't nil, it simply echos back the value, and constructs a "standardLogger" if it is nil.

Generally speaking, you'll want to try treating "New" functions like constructors in Java or other OO languages where it's always returning a new instance of something.

In the case of the logging code, however, the core logging logic need not really know or care which logging implementation is being used (similar to how an implementation of `io.Reader` utility code like `io.ReadFull(...)` need not know or care what concrete implementation is being read from).

If you can get a feel for having a good isolation of interfaces from implementations like that, then I think it'll lead to your code being more modular elsewhere in your projects.

Reply:
@cmlight thanks for the detailed feedback! Applied the changes.